### ISSUE TITLE
[Feature] Prefer destructuring in React

### ISSUE THREAD

**Issue by NishargShah** [jsx-eslint/eslint-plugin-react#3553]
In my project, there are many places where developers used `React.useState` and `useState`, so I want to make sure in future they only use `useState`.

❌ Incorrect

```jsx
const NonDestructuringComponent = () => {
  const [state, setState] = React.useState(null);

  return <p>Incorrect</p>;
};
```

✅ Correct

```jsx
import React, { useState } from 'react';

const DestructuringComponent = () => {
  const [state, setState] = useState(null);

  return <p>Correct</p>;
};
```

**Comment by ljharb** [comment]
You'd do that with `no-restricted-imports`.
